chain/chain/src/spice_core.rs

```diff
@@ -396,16 +499,28 @@ impl CoreStatementsProcessor {
             let block_hash = execution_result_original_block.get(execution_result_hash).expect(
                 "block validation should make sure that each block contains at least one endorsement for each execution_result");
 
-            if let Err(err) = tracker.try_sending_execution_result_endorsed(block_hash) {
-                // If we don't have a relevant previous block we can not be waiting for it's
-                // execution results.
-                if !matches!(err, Error::DBNotFoundErr(_)) {
-                    return Err(err);
-                }
-            }
+            tracker.try_sending_execution_result_endorsed(block_hash)?;
```

Comment:
I guess we want to send the message after store is updated? otherwise receiving actor might not yet be able to read the data from the db